### PR TITLE
Fix styling on surveyed blocks

### PR DESCRIPTION
**Pull request by kdeloach**
This fix ensures that blocks do not appear as available on the
reservations page after they have been surveyed.

Connects #1247

### CODE REVIEW

File: src/tiler/sql/user_reservable.sql [hunk 1]
```diff
@@ -12,14 +12,15 @@
     -- 'available'.  We use a GeoJSON layer in the UI for the user's
     -- reserved blockfaces, and when a blockface is deselected we want to
     -- be able to reselect it
-    WHEN reservation.user_id IS NOT DISTINCT FROM <%= user_id %> THEN 'available'
+    WHEN (block.is_available
+          AND reservation.user_id IS NOT DISTINCT FROM <%= user_id %>) THEN 'available'
```
Comment (maurizi) [inline src/tiler/sql/user_reservable.sql]: Are there known cases where a blockface can become unavailable while it is reserved by someone else?

That seems like a bug elsewhere in our code (though sanity checking here doesn't hurt).

Comment (kdeloach) [inline src/tiler/sql/user_reservable.sql]: Looks like reservations are not being deleted after submitting a survey. Not sure if that is by design or not.

Comment (maurizi) [inline src/tiler/sql/user_reservable.sql]: Ah yes of course.

Comment (kdeloach) [inline src/tiler/sql/user_reservable.sql]: Side note: Since I re-ordered these cases it might be easier to read the individual commits instead.

**Comment by jwalgran** [comment]
Looks good. Testing now.

**Comment by jwalgran** [comment]
I think these changes are good, but they do not address #1247, which has to do with the strangeness of the NYC Parks Experts group. This group has ``is_active`` set to ``False`` so that is does not appear on the public groups page. This chunk of the query prevents the ``restriction`` field from being set to `group_territory` 

```sql
    WHEN (turf.id IS NOT NULL
          AND (core_group.is_active = TRUE
               AND trustedmapper.id IS NULL
               AND core_group.admin_id <> <%= user_id %>)) THEN 'group_territory'
```

As a result, you still see the behavior described in #1247, where the line appears "available" but trusted mappers in the NYC Parks Experts group are denied access.

I think it makes sense to also remove the ``core_group.is_active = TRUE`` clause as part of this PR. "is_active" ends up being a bad name for the flag's actual purpose.

**Comment by kdeloach** [comment]
The original issue didn't mention expert mappers so I wasn't aware of that problem. I'll remove `core_group.is_active = TRUE` and then test with an "expert mappers" group next week. Thanks for reviewing.

**Comment by jwalgran** [comment]
I didn't really put it together myself until the other fixes were in.

**Comment by kdeloach** [comment]
I actually experienced a different issue. Trusted mappers in the "expert" group are able to reserve "expert" blocks... but non trusted mappers are also able to reserve these blocks.

**Comment by kdeloach** [comment]
@jwalgran Pushed a new commit to fix the other reservations styling issue.

**Comment by jwalgran** [comment]
Logged in as an individual mapper with no trusted mapper status, I still saw the NYC Parks Experts blocks colored as available. I could not reserve them, but they were blue, not grey, on the reservations map. I added two small commits on top of this PR that color the blocks as unavailable https://github.com/azavea/nyc-trees/commits/fix/jcw/reservations-style-2

I'm turning the review table on you. What do you think of my additions?

**Comment by kdeloach** [comment]
Your changes look good, but since I can't reproduce the problem as you described, I'm not able to test those changes. I'd like to get another pair of eyes on this. @RickMohr?

**Comment by RickMohr** [comment]
+1 for the combination of this PR with the extra commits from @jwalgran. As the administrator of the experts group the expert blocks were blue and I could reserve them. As a non-expert the expert blocks were gray and I could not reserve them. Blocks already mapped and blocks reserved by another user were gray and I could not reserve them.

**Comment by kdeloach** [comment]
After discussing with Rick, it looks like I didn't have any `expert_required` blocks in my system.

**Comment by kdeloach** [comment]
I cherry picked Justin's changes on top of this branch.

**Comment by kdeloach** [comment]
@azavea-bot rebuild

**Pull request merged**